pkg/securitycontext/types.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package securitycontext
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+type SecurityContextProvider interface {
+	// ModifyContainerConfig is called before the Docker createContainer call.
+	// The security context provider can make changes to the Config with which
+	// the container is created.
+	// An error is returned if it's not possible to secure the container as
+	// requested with a security context.
+	ModifyContainerConfig(pod *api.Pod, container *api.Container, config *docker.Config) error
+
+	// ModifyHostConfig is called before the Docker runContainer call.
+	// The security context provider can make changes to the HostConfig, affecting
+	// security options, whether the container is privileged, volume binds, etc.
+	// An error is returned if it's not possible to secure the container as requested
+	// with a security context.
+	ModifyHostConfig(pod *api.Pod, container *api.Container, hostConfig *docker.HostConfig)
+}
+
+// SecurityContext specifies the security constraints associated with a service account
+type SecurityContext struct {
+	// user is the uid to use when running the container
+	User int
+
+	// AllowPrivileged indicates whether this context allows privileged mode containers
+	AllowPrivileged bool
+
+	// AddCapabilities is the list of Linux kernel capabilities to add
+	AddCapabilities []string
+
+	// RemoveCapabilities is the list of Linux kernel capabilities to remove
+	RemoveCapabilities []string
+
+	// Isolation specifies the type of isolation required for containers
+	// in this security context
+	Isolation ContainerIsolationSpec
+}
+
+// ContainerIsolationSpec indicates intent for container isolation
+type ContainerIsolationSpec struct {
```

Comment:
@smarterclayton @pweil- is `ContainerIsolationSpec` or `IDMapping` the right place to have a label attribute?  We need more than just UID mapping to model MCS labels.